app/routes/synonyms/synonyms.server.ts

```diff
@@ -167,6 +154,25 @@ export async function createSynonyms(synonymousEventIds: string[]) {
   return uuid
 }
 
+export async function tryInitSynonym(eventId: string) {
+  const db = await tables()
+  const client = db._doc as unknown as DynamoDBDocument
+  const TableName = db.name('synonyms')
+  try {
+    const params = {
+      TableName,
+      Item: {
+        synonymId: { S: crypto.randomUUID() },
+        eventId: { S: eventId },
+      },
+      ConditionExpression: 'attribute_not_exists(eventId)',
+    }
+    await client.send(new PutItemCommand(params))
+  } catch (error) {
+    if ((error as Error).name !== 'ConditionalCheckFailedException') throw error
+  }
```

Comment:
I think that you can use the built-in Architect convenience function:

```ts
await db.synonyms.update({...})
```

so that you don't have to look up the actual table name, and so that the document client marshals the types for you (`{ synonymId: crypto.randomUUID() }` rather than `{ synonymId: { S: crypto.randomUUID() } }`).